modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/CalciteQueryProcessorTest.java

```diff
@@ -1148,6 +1150,22 @@ public void testSelectWithOrdering() throws IgniteInterruptedCheckedException {
             .check();
     }
 
+    /**
+     * Checks exchnage-before-correlate and correct multy-value subquery error.
+     */
+    @Test
+    public void testSubqueryMultyValueError() throws IgniteInterruptedCheckedException {
```

Comment:
The test name doesn't match javadoc. 
Actually, we should check correlated queries with correlates in projects (without correlates in filters), perhaps we should name the test (and fix the javadoc) according to this check. 
Also, fix typos (`multy` -> `multi`, `exchnage` -> `exchange`)

Reply:
Spelling fixed. Test renamed.

Filters? No filter is supposed. The best approach, I believe, would be the tree analysis. Besides, the tree is one of major goals to evolve and optimize. Probably, more effective solution here is to watch NPE/exception not to appear.